### PR TITLE
Add oidc flags for CAPI clusters

### PR DESCRIPTION
**Pull request by pipo02mix**
Add OIDC flags to the template cluster command for Cluster API cluster types.

Towards https://github.com/giantswarm/roadmap/issues/586

---

As the creator of a pull request, please consider:

- [x] Describe the goal you are trying to accomplish here, above the line.
- [x] Add a user-friendly description of your change to `CHANGELOG.md`.
- [x] Provide a link to the issue you are solving or working towards, if available.
- [x] Request SIG UX for review. They care about almost all user-facing changes.
- [x] Update the public [kubectl-gs documentation](https://docs.giantswarm.io/ui-api/kubectl-gs/) to reflect the changes here.
- [ ] add the `breaking-change` label to the PR if the change you are making changes the existing behaviour. Examples: removal of a flag, removal of a command, change of a default value. (Such changes should be released with a **major version** bump.)

Feel free to remove this checklist when done.

### CODE REVIEW

**Comment by tfussell** [comment]
Needs to be tested.

File: cmd/template/cluster/flag.go [hunk 1]
```diff
@@ -142,6 +145,11 @@ func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().StringSliceVar(&f.ControlPlaneAZ, flagControlPlaneAZ, nil, "Availability zone(s) to use by control plane nodes.")
 	cmd.Flags().StringVar(&f.Description, flagDescription, "", "User-friendly description of the cluster's purpose (formerly called name).")
 	cmd.Flags().StringVar(&f.Name, flagName, "", "Unique identifier of the cluster (formerly called ID).")
+	cmd.Flags().StringVar(&f.OIDC.IssuerURL, flagOIDCIssuerURL, "", "OIDC issuer URL.")
+	cmd.Flags().StringVar(&f.OIDC.CAFile, flagOIDCCAFile, "", "OIDC CA FilePath.")
+	cmd.Flags().StringVar(&f.OIDC.ClientID, flagOIDCClientID, "", "OIDC client ID.")
+	cmd.Flags().StringVar(&f.OIDC.UsernameClaim, flagOIDCUsernameClaim, "", "OIDC username claim.")
```
Comment (marians) [inline cmd/template/cluster/flag.go:151]: We use `email` in all installations so far, so I would welcome that as a default value here.

```suggestion
	cmd.Flags().StringVar(&f.OIDC.UsernameClaim, flagOIDCUsernameClaim, "email", "OIDC username claim.")
```

File: cmd/template/cluster/flag.go [hunk 2]
```diff
@@ -142,6 +145,11 @@ func (f *flag) Init(cmd *cobra.Command) {
...
+	cmd.Flags().StringVar(&f.OIDC.GroupsClaim, flagOIDCGroupsClaim, "", "OIDC groups claim.")
```
Comment (marians) [inline cmd/template/cluster/flag.go:152]: So far we use `groups` everywhere. Actually I doubt that this should be configurable.

```suggestion
	cmd.Flags().StringVar(&f.OIDC.GroupsClaim, flagOIDCGroupsClaim, "groups", "OIDC groups claim.")
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,11 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+
+### Added
+
+- Add OIDC flags on the template cluster command for Cluster API.
```
Comment (tfussell) [inline CHANGELOG.md:13]:
```suggestion
- Add OIDC flags to the `template cluster` command (OpenStack only).
```

File: cmd/template/cluster/flag.go [hunk 4]
```diff
@@ -159,6 +162,11 @@ func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.Description, flagDescription, "", "User-friendly description of the cluster's purpose (formerly called name).")
 	cmd.Flags().StringVar(&f.KubernetesVersion, flagKubernetesVersion, "v1.20.9", "Cluster Kubernetes version.")
 	cmd.Flags().StringVar(&f.Name, flagName, "", "Unique identifier of the cluster (formerly called ID).")
+	cmd.Flags().StringVar(&f.OIDC.IssuerURL, flagOIDCIssuerURL, "", "OIDC issuer URL.")
+	cmd.Flags().StringVar(&f.OIDC.CAFile, flagOIDCCAFile, "", "OIDC CA FilePath.")
```
Comment (tfussell) [inline cmd/template/cluster/flag.go:166]:
```suggestion
	cmd.Flags().StringVar(&f.OIDC.CAFile, flagOIDCCAFile, "", "Path to CA file used to verify OIDC issuer (optional, OpenStack only).")
```

**Review by tfussell — APPROVED** [review]

**Pull request merged**